.github/workflows/app_build.yml

```diff
@@ -0,0 +1,304 @@
+      id: cacheossdk
+      uses: actions/cache@v2
+      with:
+        path: OpenStudio-${{ env.OS_SDK_VERSION }}
+        key: OpenStudio-SDK-${{ runner.os }}-${{ env.OS_SDK_VERSION }}
+
+    - name: Download the OpenStudio installer
+      # If the SDK wasn't found in the cache
+      if: steps.cacheossdk.outputs.cache-hit != 'true'
+      # Use a bash shell so we can use the same syntax for environment variable
+      # access regardless of the host operating system
+      # Note: wget not available on windows, even in bash, so use curl
+      shell: bash
+      run: |
+        curl --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' https://api.github.com/repos/NREL/OpenStudio/releases/tags/v$OS_SDK_VERSION > os_tag.json
+        echo "PLATFORM_NAME=${{ env.PLATFORM_NAME }}"
+        cat os_tag.json
+        tar_gz_link=$(cat os_tag.json | jq -r '.assets | .[] | select(.name | contains("${{ env.PLATFORM_NAME }}")) | select(.name | contains("tar.gz")) | .browser_download_url')
+        if [ -z "$tar_gz_link" ]; then
+          echo "Could not locate the OpenStudio tar.gz for OS_SDK_VERSION=$OS_SDK_VERSION and PLATFORM_NAME=$PLATFORM_NAME"
+          exit 1;
+        fi;
+        tar_gz_name=$(basename -- $tar_gz_link)
+        tar_gz_name=$(python -c "import sys, urllib.parse as ul; print(ul.unquote_plus(\"$tar_gz_name\"))")
+        echo "Downloading $tar_gz_link"
+        echo "Normally, tar_gz_name=$tar_gz_name"
+        curl -L --output "$tar_gz_name" "$tar_gz_link"
+        ls
+        ls -la $tar_gz_name
+        echo "Extract this into the OpenStudio-$OS_SDK_VERSION directory, omitting the first directory level in the tar.gz"
+        echo "This isn't working, so use cmake to tar, then rename"
+        echo "tar xfz $tar_gz_name --strip-components 1 -C OpenStudio-$OS_SDK_VERSION"
+        echo "cmake -E make_directory OpenStudio-$OS_SDK_VERSION"
+        folder_name=${tar_gz_name%.tar.gz}
+        echo "folder_name=$folder_name"
+        cmake -E tar xfz "$tar_gz_name"
+        cmake -E rename "$folder_name" "OpenStudio-$OS_SDK_VERSION"
+
+
+    - name: Install conan
+      run: |
+          python --version
+          pip install conan
+          conan --version
+
+    - name: Cache conan cache?
+      # To avoid downloading the SDK all the time, we try to cache it
+      id: cacheconan
+      uses: actions/cache@v2
+      with:
+        path: ~/.conan
+        key: ${{ runner.os }}-conan-cache
+
+    - name: Did restoring the conan-cache work? No
+      # If the SDK wasn't found in the cache
+      if: steps.cacheconan.outputs.cache-hit != 'true'
+      run: |
+          echo "Conan cache not found"
+
+    - name: Did restoring the conan-cache work? Yes
+      # If the SDK wasn't found in the cache
+      if: steps.cacheconan.outputs.cache-hit == 'true'
+      run: |
+          ls ~/.conan/
+          ls ~/.conan/data/
+
+    - name: Cache entire build directory
```

Comment:
Caching the entire build directory might not be worth it, we tried this on local CI and found it took longer to transfer the build files than it took to rebuild.

Reply:
This is something we can adjust easily after the fact (by simply commenting it out), after we determine whether:

1. Caching actually works
2. Caching is actually beneficial

Reply:
I can see the cache being actually restored in under 3 secs, but I don't think the **build** step is actually faster because of it. Not sure why.